var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import llnl.util.tty as tty
+import os
+
+
+class Upcxx(Package):
+    """UPC++ is a C++ library that supports Partitioned Global Address Space
+    (PGAS) programming, and is designed to interoperate smoothly and
+    efficiently with MPI, OpenMP, CUDA and AMTs. It leverages GASNet-EX to
```

Comment:
There is a `gasnet` package in Spack: should this use that?

Reply:
Hi @scheibelp , thanks for reviewing this. The new UPC++ version comes with its own build system that compiles its own version of GASNet-EX in many variants under the hood. Therefore, I don't think it should use the gasnet spack package, which has probably been written for GASNet and not GASNet-EX anyway.

Reply:
> which has probably been written for GASNet and not GASNet-EX anyway

Yes that is the case. Thanks for clarifying!